http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java

```diff
@@ -41,8 +43,11 @@ private FeignClients() {}
     public static FeignClientFactory standard() {
         return FeignClientFactory.of(
                 new GuavaOptionalAwareContract(new JAXRSContract()),
-                new JacksonEncoder(ObjectMappers.guavaJdk7()),
-                new OptionalAwareDecoder(new TextDelegateDecoder(new JacksonDecoder(ObjectMappers.guavaJdk7()))),
+                new InputStreamDelegateEncoder(new JacksonEncoder(ObjectMappers.guavaJdk7())),
+                new OptionalAwareDecoder(
```

Comment:
many of the decoders/encoders are orthogonal in the sense that exactly one is going to consume and return without delegating any further. in terms of patterns, this is really more chain-of-responsibility than delegate. for the latter you would expect the delegation to walk the entire chain.